pkg/blobinfocache/internal/prioritize/prioritize.go

```diff
@@ -77,34 +82,58 @@ func (css *candidateSortState) Swap(i, j int) {
 	css.cs[i], css.cs[j] = css.cs[j], css.cs[i]
 }
 
-// destructivelyPrioritizeReplacementCandidatesWithMax is destructivelyPrioritizeReplacementCandidates with a parameter for the
-// number of entries to limit, only to make testing simpler.
-func destructivelyPrioritizeReplacementCandidatesWithMax(cs []CandidateWithTime, primaryDigest, uncompressedDigest digest.Digest, maxCandidates int) []blobinfocache.BICReplacementCandidate2 {
+// destructivelyPrioritizeReplacementCandidatesWithMax is destructivelyPrioritizeReplacementCandidates with parameters for the
+// number of entries to limit for known and unknown location separately, only to make testing simpler.
+func destructivelyPrioritizeReplacementCandidatesWithMax(cs []CandidateWithTime, primaryDigest, uncompressedDigest digest.Digest, maxCandidates int, unknownLocationLimit int) []blobinfocache.BICReplacementCandidate2 {
+	// split unknown candidates and known candidates
+	// and limit them seperately.
+	var knownLocationCandidates []CandidateWithTime
+	var unknownLocationCandidates []CandidateWithTime
+
+	for _, candidate := range cs {
+		if candidate.Candidate.UnknownLocation {
+			unknownLocationCandidates = append(unknownLocationCandidates, candidate)
+		} else {
+			knownLocationCandidates = append(knownLocationCandidates, candidate)
+		}
+	}
```

Comment:
If this is making a copy (which is fine), these functions will, AFAICS, no longer be destructive,  and it would be nice to remove the `destructively` prefix.

(Alternatively, if not renaming the functions, please add an implementation comment above this piece of code noting that these copies mean that the function is not destructive any more, to avoid confusing future readers; but that doesn’t seem to be all that much less work.)

Reply:
Added a comment will rename this private API and even the public API later on.